### ISSUE TITLE
DOM: tabbable throws an error while editor in Select Mode

### ISSUE THREAD

**Issue by ramonjd** [WordPress/gutenberg#36844]
### Description

When tabbing through elements in the Post Editor, tabbable.js throws an error when it cannot find a target element in its "focusable" array, which is an array of elements that are eligible for focus.

<img width="376" alt="Screen Shot 2021-11-25 at 3 15 08 pm" src="https://user-images.githubusercontent.com/6458278/143378760-770c1022-80af-4c5e-8f15-9efdcc677c2c.png">

The issue came to light in https://github.com/WordPress/gutenberg/pull/35655. Props to @andrewserong for first documenting it.

As far as I can see, the error is thrown in [tabbable.js](https://github.com/WordPress/gutenberg/blob/release/11.9/packages/dom/src/tabbable.js#L166) because the focussed element doesn't appear in the array of focusable elements. 

The `index` therefore is `-1`.

So when I logged things out like 

```js
	if ( index > -1 ) {
		focusables.length = index;
	} else {
		console.log( 'element index', element, index );
	}
```

The element is, more often than not, `.block-selection-button_select-button`, and this only happens when I enable select mode (hitting `Esc` in the editor). 

### Step-by-step reproduction instructions

1. Open the Post Editor, preferably with a post that has a few blocks
2. Enter Select Mode by hitting the `Esc` key. You'll know you're in Select mode when the block controls toolbar turns to black.
3. Now open the Block Editor List View.
4. Start tabbing around.

### Screenshots, screen recording, code snippet

![Kapture 2021-11-25 at 15 06 58](https://user-images.githubusercontent.com/6458278/143378269-5125f596-c1cd-44aa-b480-1ed7655c1405.gif)

### Environment info

- WordPress 5.8.1
- Gutenberg (trunk) as of November 25, 2021
- Chrome, Firefox, Safari

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alexstine** [comment]
@ramonjd Is this still valid? I thought I remembered a PR that fixed this.

**Comment by ramonjd** [comment]
> Is this still valid? I thought I remembered a PR that fixed this.

Thanks for the ping @alexstine 

Looks like this was resolved by https://github.com/WordPress/gutenberg/pull/38961
